scrapy/core/scraper.py

```diff
@@ -203,10 +203,12 @@ def call_spider(
                 callback=callback, callbackKeywords=result.request.cb_kwargs
             )
         else:  # result is a Failure
-            result.request = request
+            # TODO: properly type adding this attribute to a Failure
+            result.request = request  # type: ignore[attr-defined]
             warn_on_generator_with_return_value(spider, request.errback)
             dfd = defer_fail(result)
-            dfd.addErrback(request.errback)
+            if request.errback:
+                dfd.addErrback(request.errback)
```

Comment:
`addErrback(None)` is allowed and does nothing, but it's not documented (there is a comment saying "Default value used to be None" so maybe it was documented before but is now kept for backwards compativility) so I changed this.